### ISSUE TITLE
Bulk Merge Output Even If No Merge Takes Place

### ISSUE THREAD

**Issue by aboru** [zzzprojects/EntityFramework-Extensions#291]
### Description
I have not seen any options that would appear to support this, so I wanted to check if this is already doable or if it could be a feature in the future.

I have a collection of items which includes some portion that will map into existing entities on merge, some will be new. I have options set up so that the records that map to an existing entry with no changes involved will do nothing during the merge (no update or insert step). That much works perfectly. I need to retrieve the database Pk for all entries regardless of whether an operation was performed, currently the Pk will only come back if an update or insert was performed on the table. My current workaround is to run a second query to fetch the missing key values. If I can get these as a part of the merge without actually performing an update I can shave some excess time off of my operation which would be helpful.

### Exception
No exception involved.

### Fiddle or Project (Optional)
During the bulk merge I am sending these options:

```csharp
options.ColumnPrimaryKeyExpression = trade => new { trade.Name };
options.IgnoreOnMergeUpdateExpression = trade => new { trade.CreatedUtc };
options.ColumnOutputExpression = trade => new { trade.Pk };
options.MergeMatchedAndOneNotConditionExpression = trade => new { trade.IsDeleted };
```

I am ignoring the creation timestamp on update as I do not want to update that one ever, I am only paying attention to a soft delete flag so that I only perform an update on an existing record if the deletion state has changed.

Basically I would like to perform the output expression regardless of whether an update occurs or not as I assume the record is matched before it is ignored. This may be possible with some extra configuration but I have not been able to find anything that appears to work from the documentation.

Let me know if I can provide further details to help clarify this.

### Further technical details
- EF version: 2.1.0
- EF Extensions version: 2.7.5
- Database Provider: MS SQL Server

**Comment by JonathanMagnan** [comment]
Hello @aboru ,

There is no option that support it by default.

Is a double `BulkMerge` call to our method would be a good solution for you?

You could make the first bulk merge as you currently does without the output expression:
```csharp
context.BulkMerge(invoices, options => {
	options.ColumnPrimaryKeyExpression = trade => new { trade.Name };
	options.IgnoreOnMergeUpdateExpression = trade => new { trade.CreatedUtc };
	options.MergeMatchedAndOneNotConditionExpression = trade => new { trade.IsDeleted };
});
```

Then use the `IsReadOnly` option that will only perform the select part:

```csharp
context.BulkMerge(invoices, options => {
			options.ColumnPrimaryKeyExpression = trade => new { trade.Name };
			options.ColumnOutputExpression = trade => new { trade.Pk };
			options.IsReadOnly = true;
});
```

Or vice-versa

Let me know if that could be an accepted solution.

Best Regards,

Jonathan

**Comment by aboru** [comment]
That does work for me, I did not think of using that option but it does do what I would like it to do. 

Awesome, thank you for the help.
